packages/expo-calendar/src/next/Calendar.ts

```diff
@@ -206,32 +197,32 @@ export async function getCalendarsNext(type?: EntityTypes): Promise<ExpoCalendar
  * @param details A map of details for the calendar to be created.
  * @returns An [`ExpoCalendar`](#expocalendar) object representing the newly created calendar.
  */
-export async function createCalendarNext(details: Partial<Calendar> = {}): Promise<ExpoCalendar> {
+export function createCalendar(details: Partial<Calendar> = {}): ExpoCalendar {
   const color = details.color ? processColor(details.color) : undefined;
   const newDetails = { ...details, id: undefined, color: color || undefined };
-  const createdCalendar = await InternalExpoCalendar.createCalendarNext(newDetails);
+  const createdCalendar = InternalExpoCalendar.createCalendar(newDetails);
   Object.setPrototypeOf(createdCalendar, ExpoCalendar.prototype);
   return createdCalendar;
 }
 
 /**
  * Lists events from the device's calendar. It can be used to search events in multiple calendars.
  * > **Note:** If you want to search events in a single calendar, you can use [`ExpoCalendar.listEvents`](#listeventsstartdate-enddate) instead.
- * @param calendarIds An array of calendar IDs to search for events.
+ * @param calendars An array of calendars to search for events.
  * @param startDate The start date of the time range to search for events.
  * @param endDate The end date of the time range to search for events.
  * @returns An array of [`ExpoCalendarEvent`](#expocalendarevent) objects representing the events found.
  */
 export async function listEvents(
-  calendarIds: string[],
+  calendars: ExpoCalendar[],
```

Comment:
Hmm, to be honest, I would do it this way:
```suggestion
  calendars: string[] | ExpoCalendar[],
```
And on the JS/native side, we can map ExpoCalendar[] into the list of IDs. This way, the developers don't need to fetch all the calendars first, before fetching the events (for example, they can store the calendar IDs in some local storage).

If we do the mapping on the JS side, it wouldn't require any native changes.